stdpopsim/genomes.py

```diff
@@ -352,20 +369,49 @@ def species_contig(
                 raise ValueError("Cannot specify sequence length for named contig")
             if inclusion_mask is not None and exclusion_mask is not None:
                 raise ValueError("Cannot specify both inclusion and exclusion masks")
+
             chrom = species.genome.get_chromosome(chromosome)
+            if left is None:
+                left = 0
+            else:
+                if not 0 <= left < chrom.length:
+                    raise ValueError(
+                        f"Left coordinate {left} falls outside chromosome "
+                        f"{chrom.id} with length {chrom.length}."
+                    )
+            if right is None:
+                right = chrom.length
+            else:
+                if not left < right <= chrom.length:
+                    raise ValueError(
+                        f"Right coordinate {right} falls outside chromosome "
+                        f"{chrom.id} with length {chrom.length}."
+                    )
+
             if genetic_map is None:
                 logger.debug(f"Making flat chromosome {length_multiplier} * {chrom.id}")
                 gm = None
+                # TODO: is there a need for `length_multiplier`, given the ability
+                # to specify an arbitary interval? Combining both here results in
+                # a confusing coordinate system.
```

Comment:
Agreed. I vote to deprecate the length multiplier. It might have been useful during initial testing and development, but it doesn't compose well with any of the newer Contig features.

Reply:
I agree. So, leave the length multiplier feature in there (with a deprecation warning?) but don't try to accommodate it when adding new arguments/features.

Reply:
Yeah, exactly. Don't worry about deprecating it in this pr though.